src/app/options/options.component.ts

```diff
@@ -1,18 +1,44 @@
-import { Component, ViewChild, ElementRef } from '@angular/core';
+import { Component, ViewChild, ElementRef, OnInit } from '@angular/core';
 import { Note } from '@app/notes/notes.model';
 import { Filter, Options } from '@app/shared/model/options.model';
 import { NotesComponent } from '@app/notes/notes.component';
+import { Router, ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'app-options',
   templateUrl: './options.component.html',
 })
-export class OptionsComponent {
+export class OptionsComponent implements OnInit {
   options: Options = new Options();
   filter: Filter = new Filter();
   @ViewChild(NotesComponent, { static: true }) noteChild;
 
-  constructor() {}
+  constructor(private router: Router, private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    this.route.queryParamMap.subscribe(queryParamMap => {
+      /* tslint:disable:no-string-literal */
+      if ('filter' in this && this.filter.isEmpty()) {
+        for (const i of Object.keys(this.filter)) {
+          if (i in queryParamMap['params']) {
+            if (i !== 'markdown') {
+              if (Array.isArray(queryParamMap['params'][i])) {
+                for (const x of Object.keys(queryParamMap['params'][i])) {
+                  this.filter[i][queryParamMap['params'][i][x]] = true;
+                }
+              } else {
+                this.filter[i][queryParamMap['params'][i]] = true;
+              }
+            } else {
+              this.filter['markdown'] = queryParamMap['params']['markdown'];
```

Comment:
I would prefer in adding some setter method to the class instead of direct assignment.